packages/commonwealth/shared/schemas/createCommunitySchema.ts

```diff
@@ -0,0 +1,83 @@
+import { EnsProvider } from '@ethersproject/providers/lib/base-provider';
+import {
+  ChainBase,
+  ChainCategoryType,
+  ChainNetwork,
+  ChainType,
+} from 'common-common/src/types';
+import { z } from 'zod';
+import { ALL_COMMUNITIES } from '../../server/middleware/databaseValidationService';
+import { getFileSizeBytes } from '../../server/util/getFilesSizeBytes';
+
+export const MAX_COMMUNITY_IMAGE_SIZE_KB = 500;
+
+async function checkIconSize(val, ctx) {
+  const fileSizeBytes = await getFileSizeBytes(val);
+  if (fileSizeBytes === 0) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: "Image url provided doesn't exist",
+    });
+  }
+  if (fileSizeBytes >= MAX_COMMUNITY_IMAGE_SIZE_KB * 1024) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: `Image must be smaller than ${MAX_COMMUNITY_IMAGE_SIZE_KB}kb`,
+    });
+  }
+}
+
+export const createCommunitySchema = (provider?: EnsProvider) =>
+  z.object({
+    // visible on client params
+    name: z
+      .string()
+      .max(255)
+      .refine((data) => !data.includes(ALL_COMMUNITIES), {
+        message: `String must not contain '${ALL_COMMUNITIES}'`,
+      }),
+    community_url: z.string().url().optional(),
+    namespace: z
+      .string()
+      .refine(async (val) => {
+        if (!provider) return true;
+        return await provider.resolveName(val);
+      }, 'ENS not found')
+      .or(z.string().max(0))
+      .optional(),
+    description: z.string().optional(),
+    icon_url: z
+      .string()
+      .url()
+      .superRefine(async (val, ctx) => await checkIconSize(val, ctx))
+      .or(z.string().max(0))
+      .optional(),
+    social_links: z.array(z.string().url()).optional(),
```

Comment:
Yes, I think it is a better idea to do it right, than to continue with our current approach. The issue with our current approach is that it is not scalable. For example, if we want to add some new social link type, we need to add a whole new column to the Communities table. I think if we store social links as an array, without restrictions, we can just check on the front end if they are certain kinds of social links instead of knowing before hand on the backend. (ex. on front end, if social link starts with ://discord.com, then display discord icon beside it).

Reply:
@kurtisassad does it need any changes from the design then or we are fine with this? 
<img width="727" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/066348b1-6c86-412d-8d1f-7ef83a5cb1c9">